perf/benchmark_setup_linux.sh

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env bash
+
+# Function to determine what is set in terms of NUMA and possibly others et al
+function checkHostReadiness() {
+  echo "=========================================================="
+  echo "Running numactl --show to determine if/how NUMA is enabled"
+  echo 
+  numactl --show
+  echo "=========================================================="
+}
+
+# Function to check and echo the current SMT state
+function checkSMTState() {
+    local smt_state
+    smt_state=$(cat /sys/devices/system/cpu/smt/control)
+    echo "Current SMT State: $smt_state"
+}
+
+# Function to temporarily disable SMT
+function disableSMT() {
+    echo "============================================================="
+    echo "Temporarily Disabling SMT (Simultaneous Multithreading)"
+    
+    # Echo the SMT state before disabling
+    echo "SMT state before change:"
```

Comment:
I don't think we need to see the "before" value, just a log of the value after it's been set.